components/cam/src/physics/cam/micro_p3_interface.F90

```diff
@@ -699,6 +795,103 @@ subroutine get_cloud_fraction(its,ite,kts,kte,ast,qc,qr,qi,method, &
        return
     end subroutine get_cloud_fraction
 
+    subroutine get_prescribed_CCN(nccn_prescribed,micro_p3_lookup_dir,its,ite,kts,kte,pbuf,lchnk)
+
+       use pio,              only: file_desc_t,pio_nowrite
+       use cam_pio_utils,    only: cam_pio_openfile,cam_pio_closefile
+       use cam_grid_support, only: cam_grid_check, cam_grid_id
+       use cam_grid_support, only: cam_grid_get_dim_names
+       use ncdio_atm,        only: infld
+       use time_manager,     only: get_curr_date
+       use ppgrid,           only: begchunk, endchunk, pcols, pver, pverp, psubcols
```

Comment:
I think as long as you include all of these ppgrid variables at the header of the module you don't have to include them in individual routines.